be/src/connector/deletion_vector/deletion_vector.cpp

```diff
@@ -0,0 +1,170 @@
+                                  .fs_stats = &fs_scan_stats,
+                                  .app_stats = &app_scan_stats,
+                                  .datacache_options = _params.datacache_options};
+    ASSIGN_OR_RETURN(auto file,
+                     HdfsScanner::create_random_access_file(shared_buffered_input_stream, cache_input_stream, options));
+    std::vector<io::SharedBufferedInputStream::IORange> io_ranges{};
+    int64_t offset = _deletion_vector_descriptor->offset;
+    int64_t length = _deletion_vector_descriptor->sizeInBytes + DV_SIZE_LENGTH;
+    while (offset < length) {
+        const int64_t remain_length = std::min(static_cast<int64_t>(config::io_coalesce_read_max_buffer_size), length);
+        io_ranges.emplace_back(offset, remain_length);
+        offset += remain_length;
+    }
+
+    RETURN_IF_ERROR(shared_buffered_input_stream->set_io_ranges(io_ranges));
+    return file;
+}
+
+Status DeletionVector::deserialized_inline_dv(std::string& encoded_bitmap_data,
+                                              std::set<int64_t>* need_skip_rowids) const {
+    ASSIGN_OR_RETURN(auto decoded_bitmap_data, base85_decode(encoded_bitmap_data));
+    uint32_t inline_magic_number;
+    memcpy(&inline_magic_number, decoded_bitmap_data.data(), DeletionVector::MAGIC_NUMBER_LENGTH);
+
+    int64_t serialized_bitmap_length = decoded_bitmap_data.size() - MAGIC_NUMBER_LENGTH;
+    std::unique_ptr<char[]> deletion_vector(new char[serialized_bitmap_length]);
+    memcpy(deletion_vector.get(), decoded_bitmap_data.data() + MAGIC_NUMBER_LENGTH, serialized_bitmap_length);
+
+    return deserialized_deletion_vector(inline_magic_number, std::move(deletion_vector), serialized_bitmap_length,
+                                        need_skip_rowids);
+}
+
+Status DeletionVector::deserialized_deletion_vector(uint32_t magic_number, std::unique_ptr<char[]> serialized_dv,
+                                                    int64_t serialized_bitmap_length,
+                                                    std::set<int64_t>* need_skip_rowids) const {
+    if (magic_number != MAGIC_NUMBER) {
+        std::stringstream ss;
+        ss << "Unexpected magic number : " << magic_number;
+        return Status::RuntimeError(ss.str());
+    }
+
+    // Construct the roaring bitmap of corresponding deletion vector
+    roaring64_bitmap_t* bitmap =
+            roaring64_bitmap_portable_deserialize_safe(serialized_dv.get(), serialized_bitmap_length);
+    if (bitmap == nullptr) {
+        return Status::RuntimeError("deserialize roaring64 bitmap error");
+    }
+
+    // Construct _need_skip_rowids from bitmap
+    uint64_t bitmap_cardinality = roaring64_bitmap_get_cardinality(bitmap);
+    std::unique_ptr<uint64_t[]> bitmap_array(new uint64_t[bitmap_cardinality]);
+    roaring64_bitmap_to_uint64_array(bitmap, bitmap_array.get());
+    need_skip_rowids->insert(bitmap_array.get(), bitmap_array.get() + bitmap_cardinality);
```

Comment:
Since eventually `need_skip_rowids` will be converted to flat bitmap, we should construct the flat bitmap directly from `roaring64_bitmap_t`.

Reply:
Yes, but here we can still need this set, because the group reader need this set to find lowr/upper bound
![image](https://github.com/user-attachments/assets/4c0e9cde-9b72-47db-bf64-0d94e2daa294)